src/reducers/copy.js

```diff
@@ -0,0 +1,127 @@
+    // InsertField tulee MarcRecordista
+    if (baseFields.length === 0) {
+      sourceFields.forEach(f => base.insertField(f));
+      debug(`base after test 01: ${JSON.stringify(base, undefined, 2)}`);
+      return base;
+    }
+    // Jos basessa on jo kenttä, sourcen kenttä kopioidaan uutena basen kentän lisäksi
+    // Eli merged.jsonissa voi olla useampi kappale samaa kenttää.
+
+    // SourceField käsittelee vuorotellen jokaisen patternin mukaisen kentän sourcesta
+    const filterMissing = function(sourceField) {
+      // Testi 02: Onko identtisiä kontrollikenttiä?
+      // If value in sourceField kertoo, että kyseessä on kontrollikenttä,
+      // Koska vain kontrollikentillä on 'value'-stringi suoraan fieldissä, datakentillä joka subfieldin alla
+      if ('value' in sourceField) {
+        // Jos isIdentical palauttaa falsen, ei löydy matchia eli ei ole identtisiä kontrollikenttiä
+        // (eli koko if-blokin arvo on true)
+        debug('tarkistetaan kontrollikentät');
+        debug(`sourceField kontrollikentät on: ${JSON.stringify(sourceField, undefined, 2)}`);
+        return baseFields.some(isIdenticalControlField) === false;
+      }
+
+      // Jos sourceFieldillä on subfields-osio, kyseessä on datakenttä.
+      // Testi 04: Onko identtisiä datakenttiä? (04:ssä on 2 kpl)
+      // Testi 05: Yksi samanlainen ja yksi erilainen datakenttä
+      if ('subfields' in sourceField) {
+        debug('tarkistetaan datakentät');
+        debug(`sourceField datakentät on: ${JSON.stringify(sourceField, undefined, 2)}`);
+        return baseFields.some(isIdenticalDataField) === false;
+      }
+
+      // Debug(`filterMissingin arvo on: ${JSON.stringify(filterMissing, undefined, 2)}`);
+      // Return filterMissing; // Palauttaa ei-identtiset kentät
+
+      // Apufunktiot:
+      // Marc-kentän normalisointi
+      function normalizeControlField(field) {
+        return field.value.toLowerCase().replace(/\s+/u, '');
+      }
+      function normalizeDataField(field) {
+        // Return field.subfields[0].value.toLowerCase().replace(/\s+/u, ''); // tämä toimii mutta vain ekalle indeksille
+        const x = field.subfields.map(subfield => subfield.value.toLowerCase().replace(/\s+/u, '')); // Miksi tämä ei toimi?
+        debug(`tulos on ${x}`);
+        return x;
+      }
+
+      // Ovatko normalisoitu base ja source identtisiä?
+      // Jos funktio palauttaa true niin ovat, jos false niin eivät
+      function isIdenticalControlField(baseField) {
+        const normalizedBaseField = normalizeControlField(baseField);
+        const normalizedSourceField = normalizeControlField(sourceField);
+        return normalizedSourceField === normalizedBaseField;
+      }
+
+      // Ovatko datakentät identtisiä? tag, ind1, ind2 ja kaikki osakentät
+      function isIdenticalDataField(baseField) {
+        const normalizedBaseField = normalizeDataField(baseField);
+        const normalizedSourceField = normalizeDataField(sourceField);
```

Comment:
You don't normalize the fields, you normalize the subfields (And this can be at each iteration described below).